src/main/resources/io/viash/platforms/nextflow/WorkflowHelper.nf

```diff
@@ -603,10 +606,467 @@ def paramsToList(params, config) {
+      } else if (parValue == null) {
+          parValue = []
+      } else {
+          parValue = [ parValue ]
+      }
+      parValue = parValue.flatten()
+    }
+    // For all parameters check if multiple values are only passed for
+    // arguments that allow it
+    if (parValue instanceof Collection && !parameterSettings ) {
+      assert parValue.size() == 1 : 
+      "Error: argument ${parName} has too many values.\n" +
+      "  Expected amount: 1. Found: ${parValue.size()}"
+      parValue = parValue[0]
+    }
+    [parName, parValue]
+  }
+  return parsedParamValues
+}
+
+/**
+ * Check if the ids are unique across parameter sets
+ *
+ * @param parameterSets a list of parameter sets.
+ */
+private void _checkUniqueIds(List<Tuple2<String, Map<String, Object>>> parameterSets) {
+  def ppIds = parameterSets.collect{it[0]}
+  assert ppIds.size() == ppIds.unique().size() : "All argument sets should have unique ids. Detected ids: $ppIds"
+}
+
+/**
+ * Resolve the file paths in the parameters relative to given path
+ *
+ * @param paramList A Map containing parameters to process.
+ *                  This function assumes that files are still of type String.
+ * @param inputFileSetttings A Map of the viash configuration argument entries 
+ *                           that are considered input files (i.e. with 'direction: input'
+ *                           and 'type: file').
+ * @param relativeTo path of a file to resolve the parameters values to.
+ *
+ * @return A map of parameters where the location of the input file parameters have been resolved
+ *         resolved relatively to the provided path.
+ */
+private Map<String, Object> _resolvePathsRelativeTo(Map paramList, 
+  List<Map<String, String>> inputFileSetttings, String relativeTo) {
+  paramList.collectEntries { parName, parValue ->
+    isInputFile = inputFileSetttings.find({it.plainName == parName})
+    if (isInputFile) {
+      if (parValue instanceof List) {
+        parValue = parValue.collect({path -> 
+          path !instanceof String ? path : file(getChild(relativeTo, path))
+        })
+      } else {
+        parValue = parValue !instanceof String ? path : file(getChild(relativeTo, parValue))
+      }
+    }
+    [parName, parValue]
+  }
+}
+
+/**
+ * Parse nextflow parameters based on settings defined in a viash config 
+ * and return a nextflow channel.
+ *
+ * @param params Input parameters from nextflow.
+ * @param multiArgumentsSettings A List of the viash configuration argument entries 
+ *                               that have the property 'multiple: true'.
+ * @param inputfileArguments A List of the viash configuration argument entries 
+ *                           that specify the input files.
+ *
+ * @return A list of parameter sets that were parsed from the 'param_list' argument value.
+ */
+private List<Tuple2<String, Map>> _parseParamListArguments(Map params, 
+  List<Map> multiArguments, List<Map> inputfileArguments){
```

Comment:
Now I'm even more convinced that you should just pass the config arguments rather than make different subsets of the same argument lists for downstream checks ;)